app/hotels/src/navigation/__tests__/AllHotelsNavigationScreen.test.js

```diff
@@ -0,0 +1,59 @@
+// @flow
+
+import * as React from 'react';
+import { Platform, BackHandler } from 'react-native';
+import { shallow } from 'enzyme';
```

Comment:
Just `react-test-renderer/shallow` is not good enough?

Reply:
I crashes, don't know why. I struggle finding documentation for react-test-renderer

Reply:
Example from this repo: https://github.com/kiwicom/react-native-app/blob/2c9c102e3d65135a19db0a671dea619b2e1937dd/app/hotels/src/filter/stars/__tests__/StarsPopup.test.js#L4

Docs: https://reactjs.org/docs/shallow-renderer.html

Reply:
Sure, I will have a look at it.....

Reply:
The real issue here is that I want to test `componentDidMount` function. 
Using `react-test-renderer` crashes because a child component connects to redux store. I know with enzyme, I can pass the store like `{ context: { store } }`
But that does not work with `react-test-renderer`; 

with `react-test-renderer/shallow` it seems like I just get a snapshot (again cannot find any documentation), I can not get an instance and call `componentDidMount`

`enzyme shallow` allows me to get an instance and test `componentDidMount`

So if you have some hints or tips that can help me solve this, i would be happy :)

Reply:
> Using react-test-renderer crashes because a child component connects to redux 

Test only what you need to test and don't involve Redux. I usually export default component with HOC components and I also export the component itself just to get access to the class directly.

I am not sure if `componentDidMount` works with shallow rendering or you have to use normal renderer. But again - I usually test only what I want to test so no need to call lifecycle methods.

It's just odd that you have to pass context to the Enzym (why?) and that you need React DOM reconciler for this.

Reply:
The problem is that `react-test-renderer` calls render, and renders all child components also. I do test the exported component like `import { AllHotelsNavigationScreen } from '../AllHotelsNavigationScreen'` still it gives me `Invariant Violation: Could not find "store" in either the context or props of "Connect(FilterStripe)"` because a child of the component connects to redux store.

As for `react-dom` there is an open issue for it, https://github.com/airbnb/enzyme/issues/1390

Reply:
```
let backButtonListner = null;

export const registerListener = (onBackClicked: Function) => {
  if (Platform.OS === 'android' && this.backButtonListner === null) {
    backButtonListner = BackHandler.addEventListener(
      'hardwareBackPress',
      () => {
        onBackClicked();
        return true;
      },
    );
  }
};

export class AllHotelsNavigationScreen extends React.Component<Props> {
  

  componentDidMount = () => {
    registerListener(this.props.onBackClicked);
  };
```

Maybe  this for a workaround? So I can test the `registerListener` isolated?

Reply:
Why do you even need to render it? :)